src/mysky.ts

```diff
@@ -199,54 +236,108 @@ export class MySky {
     log("Entered getEncryptedPathSeed");
 
     // Check with the permissions provider that we have permission for this request.
-
     await this.checkPermission(path, PermCategory.Hidden, PermType.Read);
 
     // Get the seed.
-
     const seed = checkStoredSeed();
     if (!seed) {
       throw new Error("User seed not found");
     }
 
     // Compute the root path seed.
-
-    const bytes = new Uint8Array([...sha512(SALT_ENCRYPTED_PATH_SEED), ...sha512(seed)]);
-    // NOTE: Truncate to 32 bytes instead of the 64 bytes for a directory path
-    // seed. This is a historical artifact left for backwards compatibility.
-    const rootPathSeedBytes = sha512(bytes).slice(0, ENCRYPTION_ROOT_PATH_SEED_BYTES_LENGTH);
+    const rootPathSeedBytes = deriveRootPathSeed(seed);
 
     // Compute the child path seed.
-
     return deriveEncryptedPathSeedForRoot(rootPathSeedBytes, path, isDirectory);
   }
 
-  // TODO
+  /**
+   * Gets the user's preferred portal, if set.
+   *
+   * @returns - The preferred portal, if set.
+   */
+  async getPreferredPortal(): Promise<string | null> {
+    log("Entered getPreferredPortal");
+
+    return this.preferredPortal;
+  }
+
+  // TODO: Logout from all tabs.
   /**
    * Logs out of MySky.
    */
   async logout(): Promise<void> {
-    // Clear the stored seed.
+    const errors = [];
 
-    clearStoredSeed();
+    // Check if user is logged in.
+    const seed = checkStoredSeed();
```

Comment:
nit: We are doing this operation multiple times in this file. Can we avoid that by caching the seed somewhere?

Reply:
Yes, that is indeed a good point. I wrote up an issue: https://github.com/SkynetLabs/skynet-mysky/issues/243